### PR TITLE
Implemented backend tests for ssot route and rpa Task & app route

### LINKED ISSUE #150: write SSoT backend tests
As a developer, I'd like to be sure that I didn't break anything when working with the code, so I would like to have the code covered through tests.

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] All methods called that can be reached from the frontend by calling the `/ssot/` route are covered by tests
- [ ] The tests are using a local copy of a mongoDB instance to not mess with the working database

### LINKED ISSUE #151: write task+application backend tests
As a developer, I'd like to be sure that I didn't break anything when working with the code, so I would like to have the code covered through tests.
depends on #150

## :white_check_mark: &nbsp; Conditions of satisfaction

- [ ] All methods called that can be reached from the frontend by calling the `/rpa-framework/` route are covered by tests
- [ ] The tests are using a local copy of a mongoDB instance to not mess with the working database

### PR DESCRIPTION
**Pull request by SanJSp**
Fixes #150 and fixes #151

## Changes
This PR introduces some changes to enable testing of all ssot routes in our backend. It uses for this a [local mongo-db instance](https://github.com/nodkz/mongodb-memory-server) with some [mock HTTP requests](https://github.com/howardabrams/node-mocks-http), each test verifies the HTTP status and checks, whether the response contains the expected value. Furthermore tests check the database, if the object really has been created.

Additionally, I fixed some spelling issues to enforce camel-case naming of variables.

## Checklist before merge

**Developer's responsibilities**
* [ ] **Assign** one or two developers
* [ ] **Change code** if reviewer(s) has/have requested it
* [ ] **Pull request build** has passed
* [ ] **tested locally** (in at least chrome & firefox if frontend)
* [ ] updated the **documentation**
* [ ] added **tests** where necessary

### CODE REVIEW

File: server/utils/TestingUtils/TestDatabaseHandler.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  await mongod.stop();
+};
+
+/**
+ * Remove all the data for all db collections.
+ */
+module.exports.clearDatabase = async () => {
+  const { collections } = mongoose.connection;
+
+  // fix according to https://docs.w3cub.com/eslint/rules/no-await-in-loop.html
```
Comment (kej-jay) [inline server/utils/TestingUtils/TestDatabaseHandler.js:41]: does this need to be fixed or is this the fix?

File: server/routes/rpaFrameworkRoutesTest.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    expect(response.statusCode).toBe(200);
+    expect(data).toEqual([
+      testData.testRpaTask1.Application,
+      testData.testRpaTask2.Application,
+    ]);
+  });
+});
+
+describe('/commands/get-available-tasks-for-application', () => {
+  it('retreives the list of all available tasks for an application correctly', async () => {
```
Comment (kej-jay) [inline server/routes/rpaFrameworkRoutesTest.test.js:56]: typo

File: server/routes/ssotRoutesTest.test.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+/* eslint-disable no-underscore-dangle */
+const mongoose = require('mongoose');
+const httpMocks = require('node-mocks-http');
+const dbHandler = require('../utils/TestingUtils/TestDatabaseHandler');
+const ssotRetrievalController = require('../controllers/ssotRetrievalController');
+const ssotParsingController = require('../controllers/ssotParsingController');
+const testData = require('./testData');
+
+const SsotModel = mongoose.model('SSoT');
+const UaoModel = mongoose.model('userAccessObject');
```
Comment (kej-jay) [inline server/routes/ssotRoutesTest.test.js:11]: please think about not using Uao

**Review by kej-jay — CHANGES_REQUESTED** [review]
Some minor changes, mostly just typo and naming. 
grafik
Really good job! And I love the coverage!

File: server/package.json [hunk 4]
```diff
@@ -13,7 +13,9 @@
   ],
   "dependencies": {
     "express": "^4.16.4",
-    "mongoose": "^5.11.16"
+    "mongodb": "^3.6.5",
+    "mongoose": "^5.12.1",
+    "sinon": "^9.2.4"
```
Comment (WolfgangDaniel) [inline server/package.json:18]: Do we need sinon? An if so what for?

File: server/routes/ssotRoutesTest.test.js (hunk 3 above)
Comment (WolfgangDaniel) [inline server/routes/ssotRoutesTest.test.js:11]:
```suggestion
const UserAccessObjectModel = mongoose.model('userAccessObject');
```

File: server/routes/ssotRoutesTest.test.js [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+const loadSsotInDb = async () => {
+  const ssot = new SsotModel(testData.testSsot);
+  await ssot.save();
+};
+
+const loadUserAccessObjectInDb = async () => {
+  const uao = new UaoModel(testData.testUserAccessObject);
+  await uao.save();
```
Comment (WolfgangDaniel) [inline server/routes/ssotRoutesTest.test.js:20]:
```suggestion
  const userAccessObject = new UserAccessObjectModel(testData.testUserAccessObject);
  await userAccessObject.save();
```

File: server/routes/ssotRoutesTest.test.js [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    expect(response.statusCode).toBe(200);
+    expect(JSON.stringify(data.userId)).toEqual(
+      JSON.stringify(testData.userId)
+    );
+    expect(JSON.stringify(data.robotId)).toEqual(
+      JSON.stringify(testData.ssotId)
+    );
+
+    // verify if really in DB
+    const uao = await UaoModel.find({
```
Comment (WolfgangDaniel) [inline server/routes/ssotRoutesTest.test.js:173]:
```suggestion
    const userAccesObject = await UserAccesObjectModel.find({
```

File: server/routes/ssotRoutesTest.test.js [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+      userId: testData.userId,
+      robotId: testData.ssotId,
+    }).exec();
+    expect(JSON.stringify(uao[0].robotId)).toBe(
+      JSON.stringify(testData.ssotId)
+    );
+    expect(JSON.stringify(uao[0].userId)).toEqual(
```
Comment (WolfgangDaniel) [inline server/routes/ssotRoutesTest.test.js:180]:
```suggestion
    expect(JSON.stringify(userAccesObject[0].robotId)).toBe(
      JSON.stringify(testData.ssotId)
    );
    expect(JSON.stringify(userAccesObject[0].userId)).toEqual(
```

**Review by WolfgangDaniel — CHANGES_REQUESTED** [review]
Looks very good! Just some minor naming changes - you can use commit suggestion to directly implement them

File: server/utils/TestingUtils/TestDatabaseHandler.js (hunk 1 above)
Comment (SanJSp) [reply server/utils/TestingUtils/TestDatabaseHandler.js:41]: This is the fix for the es-lint no-await-in-loop error, by simply postponing the fulfillment of the promises to the end

File: server/package.json (hunk 4 above)
Comment (SanJSp) [reply server/package.json:18]: No we dont need it, will remove it

**Review by WolfgangDaniel — APPROVED** [review]

**Review by kej-jay — APPROVED** [review]

**Pull request merged**